front/app/api/file_attachments/useAddFileAttachment.ts

```diff
@@ -0,0 +1,32 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { CLErrors } from 'typings';
+
+import fetcher from 'utils/cl-react-query/fetcher';
+
+import filesKeys from './keys';
+import { IAddFileAttachmentProperties, IFileAttachment } from './types';
+
+export const addFileAttachment = async (
+  requestBody: IAddFileAttachmentProperties
+) => {
+  return fetcher<IFileAttachment>({
+    path: `/file_attachments`,
+    action: 'post',
+    body: {
+      file_attachment: { ...requestBody },
+    },
+  });
+};
+
+const useAddFileAttachment = () => {
+  const queryClient = useQueryClient();
+  return useMutation<IFileAttachment, CLErrors, IAddFileAttachmentProperties>({
+    mutationFn: addFileAttachment,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: filesKeys.lists() });
+    },
+  });
+};
+
+// ts-prune-ignore-next
```

Comment:
Including `ts-prune-ignore-next` for now, just to solve the deadcode issues. Will remove in future PRs once the hooks are in use :)